lib/crawler.js

```diff
@@ -0,0 +1,78 @@
+'use strict'
+
+const request = require('co-request')
+const emitter = require('co-emitter')
+
+class Crawler extends emitter {
+  constructor (log) {
+    super()
+
+    this.log = log('crawler')
+    this.visitedNodes = {}
+    this.visitedPairs = {}
+    this.ledgers = {}
+  }
+
+  * crawl (ledgers) {
+    Object.assign(this.ledgers, ledgers)
+    for (const ledgerPrefix in ledgers) {
+      try {
+        yield this.crawlLedger(ledgerPrefix, ledgers[ledgerPrefix])
+      } catch (err) {
+        this.log.warn('could not reach ledger "' + ledgerPrefix + '"')
+      }
+    }
+  }
+
+  * crawlLedger (ledgerPrefix, ledgerHost) {
+    const infoRes = yield request.get({uri: ledgerHost, json: true})
+    if (infoRes.statusCode !== 200) return
+    const connectors = infoRes.body.connectors
+    yield this.emit('ledger', {id: ledgerHost})
+    for (let person of connectors) {
+      // Currently only traders have the `connector` property set
+      if (person.connector) {
+        if (this.visitedNodes[person.connector]) continue
+        this.visitedNodes[person.connector] = true
+        yield this.crawlTrader(person.connector)
+      } else {
+        const nodeId = person.id
+        if (this.visitedNodes[nodeId]) continue
+        this.visitedNodes[nodeId] = true
+        yield this.emit('user', {
+          id: nodeId,
+          ledger: ledgerHost
+        })
+      }
+    }
+  }
+
+  * crawlTrader (traderHost) {
```

Comment:
Since this is being refactored, might as well change this to be "connector" instead of "trader"